src/racer/matchers.rs

```diff
@@ -658,11 +674,38 @@ fn find_doc(msrc: &str, blobend: usize) -> String {
         .collect::<Vec<_>>()  // These are needed because
         .iter()               // you cannot `rev`an `iter` that
         .rev()                // has already been `rev`ed.
-        .map(|line| if line.len() >= 4 {  // Remove "/// "
-                String::from(line[4..].to_owned())
+        .map(|line| {
+            let l = line.trim();
+            if l.len() >= 4 {  // Remove "/// "
+                String::from(l[4..].to_owned())
             } else {
                 String::new()
-            })
+            }
+        })
+        .collect::<Vec<_>>()
+        .join("\n")
+}
+
+fn find_mod_doc(msrc: &str, blobstart: usize) -> String {
+    let blob = &msrc[blobstart..];
+
+    blob.lines()
+        .take_while(|line| {
```

Comment:
Each of the blocks here calls `line.trim()`. Would it be better to have an initial `map(|line| line.trim())` to save some cycles in subsequent steps?